x/warp/contract.go

```diff
@@ -288,18 +266,31 @@ func getVerifiedWarpMessage(accessibleState contract.AccessibleState, caller com
 	// hit an error during execution.
 	unpackedPredicateBytes, err := predicateutils.UnpackPredicate(predicateBytes)
 	if err != nil {
-		return nil, remainingGas, fmt.Errorf("%w: %s", errInvalidPredicateBytes, err)
+		return nil, 0, fmt.Errorf("%w: %s", errInvalidPredicateBytes, err)
 	}
 	warpMessage, err := warp.ParseMessage(unpackedPredicateBytes)
 	if err != nil {
-		return nil, remainingGas, fmt.Errorf("%w: %s", errInvalidWarpMsg, err)
+		return nil, 0, fmt.Errorf("%w: %s", errInvalidWarpMsg, err)
+	}
+	res, err := handler.handleMessage(warpMessage)
+	if err != nil {
+		return nil, 0, err
 	}
+	return res, remainingGas, nil
+}
+
+type addressedHanlder struct{}
```

Comment:
nit: could we rename this to `addressedPayloadHandler` for consistency? (or fix it to `addressedHandler`)